src/lib/receiver.js

```diff
@@ -302,17 +319,25 @@ function createReceiver (opts) {
   })
 }
 
+// "~ipr" + base64url encoded 8-byte id
+function generateReceiverId (hmacKey) {
+  const hmac = crypto.createHmac('sha256', hmacKey)
+  hmac.update(RECEIVER_ID_PREFIX, 'utf8')
```

Comment:
Why is `RECEIVER_ID_PREFIX` used here? I would have expected a separate constant using a string like `"receiver_id"`. Also, all the constants that are used with a given HMAC key should be tracked in one place. If there is any ambiguity, it can be a very serious vulnerability.

Another option would be to have a class that wraps the HMAC key and keeps it as a private field. That way you can easily audit that class cryptographically, without having to trace everywhere the HMAC key is being passed around.